frontend/src/components/StatisticsPage/Charts.jsx

```diff
@@ -72,20 +84,70 @@ const chartOptions = {
   },
   legend: {
     display: true,
-    position: "top"
-  }
+    position: "top",
+    onClick: function(e, clickedItem) {
+        clicks += 1;
+        let chart = this.chart;
+        timeout = setTimeout(function() {
+          if(clicks === 1) {
+            defaultOnClick.call(chart, e, clickedItem);
+          }
+          else {
+            clearTimeout(timeout);
+            let index = clickedItem.datasetIndex;
+            let ci = chart;
```

Comment:
If you use arrow functions in `onClick` and `setTimeout`, you won't need reassigning `this` variables